kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,831 @@
+        print("Entering draw_gouraud_triangle")
+        assert len(points) == len(colors)
+        assert points.ndim == 3
+        assert points.shape[1] == 3
+        assert points.shape[2] == 2
+        assert colors.ndim == 3
+        assert colors.shape[1] == 3
+        assert colors.shape[2] == 4
+        shape = points.shape
+        points = points.reshape((shape[0] * shape[1], 2))
+        tpoints = trans.transform(points)
+        tpoints = tpoints.reshape(shape)
+        name = self.list_goraud_triangles.append((tpoints, colors))
+        ''' Implement this pseudocode with points and colors:
+        http://www-users.mat.uni.torun.pl/~wrona/3d_tutor/tri_fillers.html '''
+
+    def draw_gouraud_triangles(self, gc, triangles_array, colors_array,
+        transform):
+        print ("Entra en draw_gouraud_triangles")
+#         self.draw_gouraud_triangles(gc, points.reshape((1, 3, 2)),
+#                                     colors.reshape((1, 3, 4)), trans)
+
+    def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
+        x = int(x) + self.widget.x
+        y = int(y) + self.widget.y
+        if x < 0 or y < 0:
+            return
+        if ismath:
+            self.draw_mathtext(gc, x, y, s, prop, angle)
+        else:
+            font = resource_find(prop.get_name() + ".ttf")
+            if font is None:
+                plot_text = CoreLabel(font_size=prop.get_size_in_points())
+            else:
+                plot_text = CoreLabel(font_size=prop.get_size_in_points(),
+                                font_name=prop.get_name())
+            plot_text.text = "{}".format(s)
+            if prop.get_style() == 'italic':
+                plot_text.italic = True
+            if weight_as_number(prop.get_weight()) > 500:
+                plot_text.bold = True
+            plot_text.refresh()
+            with self.widget.canvas:
+                if isinstance(angle, float):
+                    PushMatrix()
+                    Rotate(angle=angle, origin=(x, y))
+                    Rectangle(pos=(x, y), texture=plot_text.texture,
+                              size=plot_text.texture.size)
+                    PopMatrix()
+                else:
+                    Rectangle(pos=(x, y), texture=plot_text.texture,
+                              size=plot_text.texture.size)
+
+    def draw_mathtext(self, gc, x, y, s, prop, angle):
+        """
+        Draw the math text using matplotlib.mathtext
+        """
+        ftimage, depth = self.mathtext_parser.parse(s, self.dpi, prop)
+        w = ftimage.get_width()
+        h = ftimage.get_height()
+        texture = Texture.create(size=(w, h))
+        texture.blit_buffer(ftimage.as_rgba_str(), colorfmt='rgba',
+                            bufferfmt='ubyte')
+        texture.flip_vertical()
+        with self.widget.canvas:
+            Rectangle(texture=texture, pos=(x, y), size=(w, h))
+
+    def draw_markers(self, gc, marker_path, marker_trans, path,
```

Comment:
Mpl says that for optim reasons `draw_markers`, `draw_path_collection`, and `draw_quad_mesh` need to be implemented, yet none of them are. No wonder it's so slow.